test/gmpTransferAdapter/erc20Transfer/executeProposal.js

```diff
@@ -0,0 +1,372 @@
+
+    FeeHandlerRouterInstance = await FeeHandlerRouterContract.new(
+      BridgeInstance.address
+    );
+    BasicFeeHandlerInstance = await BasicFeeHandlerContract.new(
+      BridgeInstance.address,
+      FeeHandlerRouterInstance.address
+    );
+
+    await BasicFeeHandlerInstance.changeFee(destinationDomainID, resourceID, fee);
+    await BridgeInstance.adminChangeFeeHandler(FeeHandlerRouterInstance.address),
+    await FeeHandlerRouterInstance.adminSetResourceHandler(
+      originDomainID,
+      resourceID,
+      BasicFeeHandlerInstance.address
+    );
+
+    GmpHandlerInstance = await GmpHandlerContract.new(BridgeInstance.address);
+    GmpTransferAdapterInstance = await GmpTransferAdapterContract.new(
+      BridgeInstance.address,
+      GmpHandlerInstance.address,
+      resourceID,
+    );
+
+    XERC20FactoryInstance = await XERC20FactoryContract.new();
+    const XERC20DeployResponse = await XERC20FactoryInstance.deployXERC20(
+      "sygmaToken",
+      "sTOK",
+      [mintingLimit],
+      [burningLimit],
+      [GmpTransferAdapterInstance.address]
+    );
+    // set XERC20 contract instance address to the address deployed via XERC20Factory
+    const deployedXERC20Address = XERC20DeployResponse.logs[0].args._xerc20
+    XERC20Instance = await XERC20Contract.at(deployedXERC20Address)
+    const lockboxDeployResponse = await XERC20FactoryInstance.deployLockbox(
+      XERC20Instance.address,
+      ERC20MintableInstance.address,
+      false
+    );
+    // set Lockbox contract instance address to the address deployed via XERC20Factory
+    const lockboxAddress = lockboxDeployResponse.logs[0].args._lockbox
+    XERC20LockboxInstance = await XERC20LockboxContract.at(lockboxAddress);
+
+    await ERC20MintableInstance.increaseAllowance(
+      XERC20LockboxInstance.address,
+      depositAmount,
+      {
+        from: depositorAddress
+      }
+    );
+    await XERC20LockboxInstance.depositTo(
+      depositorAddress,
+      depositAmount,
+      {
+        from: depositorAddress
+      }
+    );
+    await XERC20Instance.increaseAllowance(
+      GmpTransferAdapterInstance.address,
+      depositAmount,
+      {
+        from: depositorAddress
+      }
+    );
+
+    depositFunctionSignature = Helpers.getFunctionSignature(
+      GmpTransferAdapterInstance,
+      "executeProposal"
+    );
+
+    const GmpHandlerSetResourceData =
+    Helpers.constructGenericHandlerSetResourceData(
+      depositFunctionSignature,
+      Helpers.blankFunctionDepositorOffset,
+      Helpers.blankFunctionSig
+    );
+
+    await BridgeInstance.adminSetResource(
+      GmpHandlerInstance.address,
+      resourceID,
+      GmpHandlerInstance.address,
+      GmpHandlerSetResourceData
+    );
+
+    const preparedExecutionData = await GmpTransferAdapterInstance.prepareDepositData(
+      recipientAddress,
+      XERC20Instance.address,
+      transferredAmount
+    );
+    depositData = Helpers.createGmpDepositData(
```

Comment:
This generates the deposit data correctly, but the adapter itself does it in the wrong way.